cmd/discord-provider/server/template.go

```diff
@@ -0,0 +1,19 @@
+package server
+
+// Necessary struct to send a embedded discord message
+type Message struct {
+	Username  string     `json:"username,omitempty"`
+	Content   string     `json:"content,omitempty"`
+	Embeddeds []Embedded `json:"embeds,omitempty"`
+}
+
+type Embedded struct {
+	Title       string `json:"title,omitempty"`
+	Description string `json:"description,omitempty"`
+	Color       string `json:"color,omitempty"`
+	Footer      Footer `json:"footer,omitempty"`
+}
+
+type Footer struct {
+	Text string `json:"text,omitempty"`
+}
```

Comment:
since these 3 strucs are only used within the `server` package, their names can be lowercased (i.e: Message -> message)